lib/patchelf/patcher.rb

```diff
@@ -19,13 +19,27 @@ class Patcher
     # @param [String] filename
     #   Filename of input ELF.
     # @param [Boolean] logging
+    #   *deprecated*
     #   Whether to use a (stderr-initialized) logger for errors
-    def initialize(filename, logging: true)
+    # @param [:log, :silent, :exception] on_error
+    #   action when the desired segment/tag field isn't present
+    #     :log = logs to stderr
+    #     :exception = raise exception related to the error
+    #     :silent = ignore the errors
+    #
+    # @raise [ArgumentError] when both logging and on_error are passed to initializer
+    def initialize(filename, on_error: :log, logging: nil)
+      # not foolproof, doesn't catch when (on_error: :log) is explicitly passed.
+      raise ArgumentError, "on_error and logging can't be passed together" if !logging.nil? && on_error != :log
```

Comment:
Actually my point is don't care `logging` anymore (except `@on_error = !logging ? :exception : on_error` for compatibility reason). So even the case `!logging.nil? && on_error != :log` is not needed, same as the `@raise` in doc.